File: electron-app/scripts/generate-icons.py
#!/usr/bin/env python3

from PIL import Image, ImageDraw, ImageFont
import os

print("🎨 Generating StreamVoice icons...")

# Icon sizes needed
sizes = [16, 32, 48, 64, 128, 256]

# Create assets directory if it doesn't exist
assets_dir = os.path.join(os.path.dirname(__file__), '..', 'assets')
os.makedirs(assets_dir, exist_ok=True)

for size in sizes:
    # Create new image with gradient background
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)

    # Draw rounded rectangle with gradient effect
    corner_radius = int(size * 0.2)

    # Purple gradient background (simplified - just solid color)
    purple = (118, 75, 162, 255)  # #764ba2

    # Draw rounded rectangle
    x0, y0 = 0, 0
    x1, y1 = size - 1, size - 1

    # Main rectangle
    draw.rounded_rectangle([x0, y0, x1, y1], radius=corner_radius, fill=purple)

    # Draw microphone icon (simplified)
    white = (255, 255, 255, 255)

    # Microphone body
    mic_width = int(size * 0.3)
    mic_height = int(size * 0.4)
    mic_x = (size - mic_width) // 2
    mic_y = int(size * 0.2)

    # Draw mic oval
    draw.ellipse([mic_x, mic_y, mic_x + mic_width, mic_y + mic_height], fill=white)

    # Draw stand
    stand_y = mic_y + mic_height
    draw.arc([mic_x - mic_width//4, stand_y - 5, mic_x + mic_width + mic_width//4, stand_y + mic_height//2],
             0, 180, fill=white, width=max(2, size//32))

    # Vertical line
    draw.line([(size//2, stand_y + mic_height//4), (size//2, size - size//5)],
              fill=white, width=max(2, size//32))

    # Base line
    base_width = int(size * 0.4)
    base_x = (size - base_width) // 2
    base_y = size - size//5
    draw.line([base_x, base_y, base_x + base_width, base_y],
              fill=white, width=max(3, size//24))

    # Save PNG
    img.save(os.path.join(assets_dir, f'icon-{size}.png'))
    print(f"✅ Created icon-{size}.png")

# Create main icon.png (copy of 256)
img_256 = Image.open(os.path.join(assets_dir, 'icon-256.png'))
img_256.save(os.path.join(assets_dir, 'icon.png'))
print("✅ Created icon.png")

# Create small icon (copy of 32)
img_32 = Image.open(os.path.join(assets_dir, 'icon-32.png'))
img_32.save(os.path.join(assets_dir, 'icon-small.png'))
print("✅ Created icon-small.png")

# Create tray icon (16x16 white on transparent)
tray_img = Image.new('RGBA', (16, 16), (0, 0, 0, 0))
tray_draw = ImageDraw.Draw(tray_img)

# Simple mic for tray
tray_draw.ellipse([5, 2, 11, 8], fill=white)
tray_draw.arc([3, 7, 13, 13], 0, 180, fill=white, width=1)
tray_draw.line([(8, 10), (8, 14)], fill=white, width=1)
tray_draw.line([(5, 14), (11, 14)], fill=white, width=1)

tray_img.save(os.path.join(assets_dir, 'tray-icon.png'))
print("✅ Created tray-icon.png")

# Create ICO file (multi-resolution Windows icon)
# For now, just use the 256px version
img_256.save(os.path.join(assets_dir, 'icon.ico'), format='ICO', sizes=[(256, 256)])
print("✅ Created icon.ico")

print("\n🎉 All icons generated successfully!")

File: electron-app/scripts/generate-icons.js
const { createCanvas, loadImage } = require('canvas');
const fs = require('fs');
const path = require('path');

// Icon sizes needed for Electron
const sizes = [16, 32, 48, 64, 128, 256];

async function generateIcons() {
    console.log('🎨 Generating app icons...');

    // For now, create simple gradient icons
    for (const size of sizes) {
        const canvas = createCanvas(size, size);
        const ctx = canvas.getContext('2d');

        // Create gradient background
        const gradient = ctx.createLinearGradient(0, 0, size, size);
        gradient.addColorStop(0, '#667eea');
        gradient.addColorStop(1, '#764ba2');

        // Draw rounded rectangle
        const radius = size * 0.25;
        ctx.beginPath();
        ctx.moveTo(radius, 0);
        ctx.lineTo(size - radius, 0);
        ctx.quadraticCurveTo(size, 0, size, radius);
        ctx.lineTo(size, size - radius);
        ctx.quadraticCurveTo(size, size, size - radius, size);
        ctx.lineTo(radius, size);
        ctx.quadraticCurveTo(0, size, 0, size - radius);
        ctx.lineTo(0, radius);
        ctx.quadraticCurveTo(0, 0, radius, 0);
        ctx.closePath();
        ctx.fillStyle = gradient;
        ctx.fill();

        // Draw microphone icon
        ctx.fillStyle = 'white';
        ctx.strokeStyle = 'white';
        ctx.lineWidth = size * 0.03;

        // Microphone body
        const micWidth = size * 0.2;
        const micHeight = size * 0.35;
        const micX = (size - micWidth) / 2;
        const micY = size * 0.15;

        // Draw mic shape
        ctx.beginPath();
        ctx.arc(size / 2, micY + micWidth / 2, micWidth / 2, Math.PI, 0);
        ctx.lineTo(micX + micWidth, micY + micHeight - micWidth / 2);
        ctx.arc(size / 2, micY + micHeight - micWidth / 2, micWidth / 2, 0, Math.PI);
        ctx.closePath();
        ctx.fill();

        // Stand arc
        ctx.beginPath();
        ctx.arc(size / 2, micY + micHeight, size * 0.15, 0, Math.PI, true);
        ctx.stroke();

        // Stand base
        ctx.beginPath();
        ctx.moveTo(size / 2, micY + micHeight + size * 0.15);
        ctx.lineTo(size / 2, size * 0.75);
        ctx.stroke();

        ctx.beginPath();
        ctx.moveTo(size * 0.35, size * 0.75);
        ctx.lineTo(size * 0.65, size * 0.75);
        ctx.stroke();

        // Save PNG
        const buffer = canvas.toBuffer('image/png');
        fs.writeFileSync(path.join(__dirname, '..', 'assets', `icon-${size}.png`), buffer);
        console.log(`✅ Created icon-${size}.png`);
    }

    // Create ICO file for Windows (placeholder for now)
    const iconPath = path.join(__dirname, '..', 'assets', 'icon.ico');
    // Copy largest PNG as ICO placeholder
    fs.copyFileSync(
        path.join(__dirname, '..', 'assets', 'icon-256.png'),
        iconPath
    );
    console.log('✅ Created icon.ico');

    // Create tray icons (smaller)
    const trayCanvas = createCanvas(16, 16);
    const trayCtx = trayCanvas.getContext('2d');

    // Simple mic icon for tray
    trayCtx.fillStyle = 'white';
    trayCtx.fillRect(6, 2, 4, 8);
    trayCtx.strokeStyle = 'white';
    trayCtx.lineWidth = 1;
    trayCtx.beginPath();
    trayCtx.arc(8, 10, 3, 0, Math.PI, true);
    trayCtx.stroke();
    trayCtx.beginPath();
    trayCtx.moveTo(8, 13);
    trayCtx.lineTo(8, 15);
    trayCtx.stroke();

    fs.writeFileSync(
        path.join(__dirname, '..', 'assets', 'tray-icon.png'),
        trayCanvas.toBuffer('image/png')
    );
    console.log('✅ Created tray-icon.png');

    // Also save as icon.png (copy of 256)
    fs.copyFileSync(
        path.join(__dirname, '..', 'assets', 'icon-256.png'),
        path.join(__dirname, '..', 'assets', 'icon.png')
    );

    // Save small icon
    fs.copyFileSync(
        path.join(__dirname, '..', 'assets', 'icon-32.png'),
        path.join(__dirname, '..', 'assets', 'icon-small.png')
    );

    console.log('🎉 All icons generated!');
}

// Run if called directly
if (require.main === module) {
    generateIcons().catch(console.error);
}

module.exports = { generateIcons };